src/contracts/GPv2SettlementSimulator.sol

```diff
@@ -0,0 +1,401 @@
+            abi.encode(
+                DOMAIN_TYPE_HASH,
+                DOMAIN_NAME,
+                DOMAIN_VERSION,
+                chainId,
+                address(this)
+            )
+        );
+        allowanceManager = new GPv2AllowanceManager();
+    }
+
+    // solhint-disable-next-line no-empty-blocks
+    receive() external payable {
+        // NOTE: Include an empty receive function so that the settlement
+        // contract can receive Ether from contract interactions.
+    }
+
+    /// @dev This modifier is called by settle function to block any non-listed
+    /// senders from settling batches.
+    modifier onlySolver {
+        require(authenticator.isSolver(msg.sender), "GPv2: not a solver");
+        _;
+    }
+
+    /// @dev Settle the specified orders at a clearing price. Note that it is
+    /// the responsibility of the caller to ensure that all GPv2 invariants are
+    /// upheld for the input settlement, otherwise this call will revert.
+    /// Namely:
+    /// - The fee factor cannot lead to fees > 0.1%
+    /// - All orders are valid and signed
+    /// - Accounts have sufficient balance and approval.
+    /// - Settlement contract has sufficient balance to execute trades. Note
+    ///   this implies that the accumulated fees held in the contract can also
+    ///   be used for settlement. This is OK since:
+    ///   - Solvers need to be authorized
+    ///   - Misbehaving solvers will be slashed for abusing accumulated fees for
+    ///     settlement
+    ///   - Critically, user orders are entirely protected
+    ///
+    /// Note that some parameters are encoded as packed bytes in order to save
+    /// calldata gas. For more information on encoding format consult the
+    /// [`GPv2Encoding`] library.
+    ///
+    /// @param tokens An array of ERC20 tokens to be traded in the settlement.
+    /// Orders and interactions encode tokens as indices into this array.
+    /// @param clearingPrices An array of clearing prices where the `i`-th price
+    /// is for the `i`-th token in the [`tokens`] array.
+    /// @param encodedTrades Encoded trades for signed EOA orders.
+    /// @param encodedInteractions Encoded smart contract interactions.
+    /// @param encodedOrderRefunds Encoded order refunds for clearing storage
+    /// related to invalid orders.
+    function settle(
+        IERC20[] calldata tokens,
+        uint256[] calldata clearingPrices,
+        bytes calldata encodedTrades,
+        bytes calldata encodedInteractions,
+        bytes calldata encodedOrderRefunds
+    ) external onlySolver {
+        GPv2TradeExecutionWithErrorMsg.Data[] memory executedTrades =
+            computeTradeExecutions(tokens, clearingPrices, encodedTrades);
+        allowanceManager.transferIn2(executedTrades);
+
+        executeInteractions(encodedInteractions);
+
+        transferOut(executedTrades);
+
+        claimOrderRefunds(encodedOrderRefunds);
+        revert("function can be called");
```

Comment:
The clone must revert in the settlement function at the end. Its not intended for execution